### PR TITLE
Diagnostics: Improved highlighting from parse tree

### PR DESCRIPTION
**Pull request by KevinEady**
Use source parse tree for highlighting of identifiers instead of linear token list.

### CODE REVIEW

File: pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp [hunk 1]
```diff
@@ -20,7 +15,182 @@ void SemanticTokensBuilder::build()
...
+  visitChildren( ctx );
+  return antlrcpp::Any();
+}
+
+antlrcpp::Any SemanticTokensBuilder::visitForeachIterableExpression(
+    EscriptParser::ForeachIterableExpressionContext* ctx )
+{
+  if ( auto identifier = ctx->IDENTIFIER() )
+  {
+    define( workspace, identifier->getSymbol(), SemanticTokenType::NAMESPACE );
```
Comment (KevinEady) [inline pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp:54]: wrong, VARIABLE

File: pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp [hunk 2]
```diff
@@ -20,7 +15,182 @@ void SemanticTokensBuilder::build()
...
+  visitChildren( ctx );
+  return antlrcpp::Any();
+}
+
+antlrcpp::Any SemanticTokensBuilder::visitStringIdentifier(
+    EscriptParser::StringIdentifierContext* ctx )
+{
+  if ( auto identifier = ctx->IDENTIFIER() )
+  {
+    define( workspace, identifier->getSymbol(), SemanticTokenType::NAMESPACE );
```
Comment (KevinEady) [inline pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp:43]: add comment to method why it is namespace

File: pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp [hunk 3]
```diff
@@ -20,7 +15,182 @@ void SemanticTokensBuilder::build()
...
+{
+  define( workspace, ctx->IDENTIFIER()->getSymbol(), SemanticTokenType::VARIABLE );
+  visitChildren( ctx );
+  return antlrcpp::Any();
+}
+
+antlrcpp::Any SemanticTokensBuilder::visitProgramParameter(
+    EscriptParser::ProgramParameterContext* ctx )
+{
+  define( workspace, ctx->IDENTIFIER()->getSymbol(), SemanticTokenType::VARIABLE );
```
Comment (KevinEady) [inline pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp:111]: wrong, PARAMETER

File: pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp [hunk 4]
```diff
@@ -20,7 +15,182 @@ void SemanticTokensBuilder::build()
...
+  visitChildren( ctx );
+  return antlrcpp::Any();
+}
+
+antlrcpp::Any SemanticTokensBuilder::visitFunctionParameter(
+    EscriptParser::FunctionParameterContext* ctx )
+{
+  define( workspace, ctx->IDENTIFIER()->getSymbol(), SemanticTokenType::VARIABLE );
```
Comment (KevinEady) [inline pol-core/bscript/compiler/analyzer/SemanticTokensBuilder.cpp:119]: wrong, PARAMETER

**Review by frozenblit — APPROVED** [review]

**Pull request merged**